cmd/download-plugins/main.go

```diff
@@ -184,11 +184,11 @@ func downloadReleaseToDir(ctx context.Context, client *http.Client, plugin relea
 		}
 	}()
 	log.Printf("downloading: %v", plugin.URL)
-	req, err := http.NewRequestWithContext(ctx, http.MethodGet, plugin.URL, nil)
+	req, err := client.NewRequest(http.MethodGet, plugin.URL, nil)
 	if err != nil {
 		return fmt.Errorf("failed to make HTTP request: %w", err)
 	}
-	resp, err := client.Do(req)
+	resp, err := client.BareDo(ctx, req)
```

Comment:
Do we have to handle this now?

> If rate limit is exceeded and reset time is in the future, BareDo returns *RateLimitError immediately without making a network API call.

Reply:
I think this is only if we would've ignored a previous rate limit in another request.

With `SleepUntilPrimaryRateLimitResetWhenRateLimited` enabled, it should wait until the request can be fulfilled so we won't have to worry about this.

Reply:
Okay, I guess I don't get why do _both_ changes?

Afaics we've done what is expected in the previous PR, i.e., relating to https://github.com/google/go-github/pull/3117

Reply:
If you look at the tests from that PR, it is using the GitHub client, not the http.Client. I believe if we make calls on the http.Client, we miss some of these behaviors (handling errors, retrying requests, etc.).

Reply:
Sure, my only question is why switching from `.Do` to `.BareDo`

Reply:
We could use `.Do` just would have to move around the writer, remove the body close, etc. I can do that if you'd prefer.